src/main/java/org/myrobotlab/service/interfaces/TextPublisher.java

```diff
@@ -1,22 +1,48 @@
 package org.myrobotlab.service.interfaces;
 
-public interface TextPublisher {
+import org.myrobotlab.framework.interfaces.NameProvider;
 
-  public String getName();
+public interface TextPublisher extends NameProvider {
 
-  public String publishText(String text);
+  /**
+   * These are all the methods that the utterance publisher should produce.
+   */
+  public static String[] publishMethods = new String[] { "publishText" };
 
-  @Deprecated /* use standard attachTextListener */
-  public void addTextListener(TextListener service);
+  /**
+   * Define the methods that an utterance publisher should have
+   * 
+   * @param text
+   * @return
+   */
+  public String publishText(String text);
 
-  public void attachTextListener(TextListener service);
+  /**
+   * Attach a text listener
+   * 
+   * @param service
+   */
+  default public void attachTextListener(TextListener service) {
+    attachTextListener(service.getName());
+  }
 
   /**
-   * the root for any other attachTextListener type
+   * Default way to attach an utterance listener so implementing classes need
+   * not worry about these details.
    * 
    * @param name
-   *          - the name of the text listener
    */
-  public void attachTextListener(String name);
+  default public void attachTextListener(String name) {

```

Comment:
please add removeTextListener for completeness of this interface.

Reply:
you probably mean detachTextListener .. but ya I agree

Reply:
not part of my pr ... but I'll do the penance because i really like the pattern :)
updated